eng/common/TestResources/New-TestResources.ps1

```diff
@@ -200,35 +200,62 @@ try {
         # Make sure the user is logged in to create a service principal.
         $context = Get-AzContext;
         if (!$context) {
-            $subscriptionName = $SubscriptionId
 
-            # Use cache of well-known team subs without having to be authenticated.
-            $wellKnownSubscriptions = @{
-                'faa080af-c1d8-40ad-9cce-e1a450ca5b57' = 'Azure SDK Developer Playground'
-                'a18897a6-7e44-457d-9260-f2854c0aca42' = 'Azure SDK Engineering System'
-                '2cd617ea-1866-46b1-90e3-fffb087ebf9b' = 'Azure SDK Test Resources'
+            # Use the given subscription ID if provided.
+            $subscriptionArgs = if ($SubscriptionId) {
+                @{Subscription = $SubscriptionId}
+            } else {
+                @{}
             }
 
-            if ($wellKnownSubscriptions.ContainsKey($SubscriptionId)) {
-                $subscriptionName = '{0} ({1})' -f $wellKnownSubscriptions[$SubscriptionId], $SubscriptionId
+            # TODO: Should we default DeleteAfterHours for the Playground sub to 48hrs or even a week? Could really help with costs.
+            # Or should we prompt - default and ask to opt out of DeleteAfterHours?
+
+            Log 'Logging you in...'
+            $context = (Connect-AzAccount @subscriptionArgs).Context
+
+            # Warn that a "random" subscription was selected.
+            if (!$SubscriptionId) {
+                $subscriptionName = $context.Subscription.Id
+
+                # Use cache of well-known team subs without having to be authenticated.
+                $wellKnownSubscriptions = @{
+                    'faa080af-c1d8-40ad-9cce-e1a450ca5b57' = 'Azure SDK Developer Playground'
+                    'a18897a6-7e44-457d-9260-f2854c0aca42' = 'Azure SDK Engineering System'
+                    '2cd617ea-1866-46b1-90e3-fffb087ebf9b' = 'Azure SDK Test Resources'
+                }
+
+                if ($wellKnownSubscriptions.ContainsKey($subscriptionName)) {
+                    $subscriptionName = '{0} ({1})' -f $wellKnownSubscriptions[$subscriptionName], $subscriptionName
+                }
+
+                Write-Warning "No subscription was specified, so $subscriptionName was automatically selected."
```

Comment:
Might be helpful to surround the sub name with `` `"...`" `` so the log is easier to read since the name contains spaces.